articles/ros2_security_enclaves.md

```diff
@@ -0,0 +1,315 @@
+By convention, ros2launch could be used to prefix relative enclave paths for single process node or containers using the namespace in scope, to enable a convention of composable launch files with adjustable and parameterized enclave paths.
+Given the runtime command argument for specifying the fully qualified enclave path, ros2launch would accordingly resolve relative enclave paths for executables, as defined by launch attributes.
+
+### Unqualified enclave path
+
+For single process nodes with unqualified enclave paths, the enclave directory will subsequently default to the root level enclave.
+
+``` xml
+<launch>
+  <node pkg="demo_nodes_cpp" exec="talker"/>
+  <node pkg="demo_nodes_cpp" exec="listener"/>
+</launch>
+```
+
+```
+$ tree enclaves/
+enclaves/
+├── cert.pem
+├── governance.p7s
+├── identity_ca.cert.pem -> ../public/identity_ca.cert.pem
+├── key.pem
+├── permissions_ca.cert.pem -> ../public/permissions_ca.cert.pem
+└── permissions.p7s
+```
+
+### Pushed unqualified enclave path
+
+For single process nodes with unqualified enclave paths pushed by a namespace, the enclave directory will subsequently be pushed to the relative sub-folder.
+
+``` xml
+<launch>
+  <node pkg="demo_nodes_cpp" exec="talker"/>
+  <group>
+    <push_ros_namespace namespace="foo"/>
+    <node pkg="demo_nodes_cpp" exec="listener"/>
+  </group>
+</launch>
+```
+
+```
+$ tree --dirsfirst enclaves/
+enclaves/
+├── foo
+│   ├── cert.pem
+│   ├── governance.p7s
+│   ├── identity_ca.cert.pem
+│   ├── key.pem
+│   ├── permissions_ca.cert.pem
+│   └── permissions.p7s
+├── cert.pem
+├── governance.p7s
+├── identity_ca.cert.pem
+├── key.pem
+├── permissions_ca.cert.pem
+└── permissions.p7s
+```
+
+> Symbolic links suppressed for readability
+
+### Relatively pushed qualified enclave path
+
+For single process nodes with qualified enclave paths pushed by a namespace, the qualified enclave directory will subsequently be pushed to the relative sub-folder.
+
+``` xml
+<launch>
+  <group>
+    <push_ros_namespace namespace="foo"/>
+    <node pkg="demo_nodes_cpp" exec="listener" enclave="bar"/>
+  </group>
+</launch>
+```
+
+```
+$ tree --dirsfirst enclaves/
+enclaves/
+└── foo
+    └── bar
+        ├── cert.pem
+        ├── governance.p7s
+        ├── identity_ca.cert.pem
+        ├── key.pem
+        ├── permissions_ca.cert.pem
+        └── permissions.p7s
+```
+
+### Fully qualified enclave path
+
+For single process nodes with absolute enclave paths, namespaces do not subsequently push the relative sub-folder.
+
+``` xml
+<launch>
+  <group>
+    <push_ros_namespace namespace="foo"/>
+    <node pkg="demo_nodes_cpp" exec="listener" enclave="/bar"/>
+  </group>
+</launch>
+```
+
+```
+$ tree --dirsfirst enclaves/
+enclaves/
+└── bar
+    ├── cert.pem
+    ├── governance.p7s
+    ├── identity_ca.cert.pem
+    ├── key.pem
+    ├── permissions_ca.cert.pem
+    └── permissions.p7s
+```
+
+
+## Alternatives
+
+### Context path orthogonal to namespace
```

Comment:
This alternative is the current situation.

Reply:
Done: e4b4ad4